src/states/common/base.rs

```diff
@@ -245,20 +232,28 @@ pub trait Base: Display {
 
     fn handle_new_deserialised_message(
         &mut self,
-        pub_id: PublicId,
+        src_addr: SocketAddr,
         message: Message,
         outbox: &mut EventBox,
     ) -> Result<Transition, RoutingError> {
         match message {
-            Message::Hop(msg) => self.handle_hop_message(msg, pub_id, outbox),
+            Message::Hop(msg) => self.handle_hop_message(msg, outbox),
```

Comment:
Wouldn't it be better to try and get the public ID from the PeerMap?

Or maybe we should make the `HopMessage` a variant of `DirectMessage`? This way we would get the ID of the previous hop from the `SignedDirectMessage`... and hop messages _are_ passed directly between nodes, anyway.

The workaround with using `sent_to` is kind of okay, but we're removing `sent_to` as part of RMD, so it will cause problems again...

Reply:
> Wouldn't it be better to try and get the public ID from the PeerMap?

We might not necessarily have it. We only have the public id if we received at least one `DirectMessage` from them previously, or if we received `ConnectRequest`. Not sure that's guaranteed to happen. Maybe it is. This might need more thinking through...